crypto.go

```diff
@@ -171,11 +171,14 @@ const onionErrorLength = 2 + 2 + 256 + sha256.Size
 // DecryptError attempts to decrypt the passed encrypted error response. The
 // onion failure is encrypted in backward manner, starting from the node where
 // error have occurred. As a result, in order to decrypt the error we need get
-// all shared secret and apply decryption in the reverse order.
-func (o *OnionErrorDecrypter) DecryptError(encryptedData []byte) (*btcec.PublicKey, []byte, error) {
+// all shared secret and apply decryption in the reverse order. An integer is
+// returned that represents the position of the error node in the path. Index
+// zero is the self node. Using an integer allows to distinguish between errors
+// from nodes that occur in the path multiple times.
+func (o *OnionErrorDecrypter) DecryptError(encryptedData []byte) (int, []byte, error) {
```

Comment:
With a little more work, I think the caller can just check the pointer in the main backing array to figure out which exact instance this is. Alternatively, we can return both the index and the public key. IMO, API wise, it's a little _too_ bare to just return the index without any public key as well. We also have other consumers of this library other than us as well.

Reply:
I don't think relying on the pointer still pointing to the same element is good api design. Also imo just returning an index isn't too bare, it is exactly what is needed. No redundancy.

I compare it with:
`var index = strings.Index(str, "Golang")`

Reply:
Also returning the index pushes users of the API towards considering the scenario where a node is visited in the path twice. By just returning the pubkey, it is easy to assume that it relates 1-on-1 to a position in the path. We oversaw this ourselves even.